### PR TITLE
[feat] Add dropout_rate as an optional arg

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
Adds `dropout_rate` as an optional arg to `Whisper` and `whisper.load_model`.

Currently we override the dropout for the MLP manually in our training script so as not to change the `nn.Sequential` module.

See: https://github.com/sanchit-gandhi/seq2seq-speech/pull/91/commits/dbaf79b55b3ae2e96342f6eaca5321144ee0000d for the corresponding training script changes.

### CODE REVIEW

File: whisper/model.py [hunk 1]
```diff
@@ -135,22 +136,33 @@ def forward(
...
         super().__init__()
+        self.dropout = dropout
+
         self.conv1 = Conv1d(n_mels, n_state, kernel_size=3, padding=1)
         self.conv2 = Conv1d(n_state, n_state, kernel_size=3, stride=2, padding=1)
         self.register_buffer("positional_embedding", sinusoids(n_ctx, n_state))
 
         self.blocks: Iterable[ResidualAttentionBlock] = nn.ModuleList(
-            [ResidualAttentionBlock(n_state, n_head) for _ in range(n_layer)]
+            [ResidualAttentionBlock(n_state, n_head, dropout=self.dropout) for _ in range(n_layer)]
```
Comment (patrickvonplaten) [inline whisper/model.py:165]: I wouldn't add dropout here because we freeze the encoder during training -> let's maybe disable it here?

File: whisper/model.py [hunk 2]
```diff
@@ -135,22 +136,33 @@ def forward(
...
+        # instead, we add dropout by overriding our pre-trained model in our training script (much faster!)
         x = x + self.mlp(self.mlp_ln(x))
         return x
 
 
 class AudioEncoder(nn.Module):
-    def __init__(self, n_mels: int, n_ctx: int, n_state: int, n_head: int, n_layer: int):
+    def __init__(self, n_mels: int, n_ctx: int, n_state: int, n_head: int, n_layer: int, dropout: float):
         super().__init__()
+        self.dropout = dropout
```
Comment (patrickvonplaten) [inline whisper/model.py]:
```suggestion
        self.dropout = 0.0
```

Comment (patrickvonplaten) [reply whisper/model.py]: Encoder should not use dropout
